packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -1,11 +1,19 @@
 import path from 'path';
 import type * as ts from 'typescript';
 
-interface FileSpecifier {
+type FileSpecifier = {
   from: 'file';
   name: string | string[];
-  path?: string;
-}
+} & (
+  | {
+      path?: string;
+      excludePaths?: undefined;
+    }
+  | {
+      path?: undefined;
+      excludePaths?: string | string[]; // defaults to `["node_modules"]`.
```

Comment:
I also wonder if we should call this `pathExcludes`, to visually lump it in with `path`... 🤔